src/clj/vanilla/url_handlers.clj

```diff
@@ -25,40 +25,124 @@
+    (log/info "SAVE-LAYOUT Handler(vec-ord): " vec-ord)
+    (log/info (first vec-ord))
+    (prn sorted-vals)
+    (db/save-layout! db/vanilla-db {:layout sorted-vals})))
+
+(comment
+
+  (def data
+    ({:ret_types [:data-format/x-y :data-format/x-y-n :data-format/x-y-e :data-format/y],
+      :key "adade292-295c-4ee7-9cc4-48f2dd
+224d1a", :name :area-widget, :username "chad1",
+      :basis :chart, :data-source :bubble-service, :type :area-chart,
+      :icon "/images/area-widget.png", :label "Area",
+      :data-grid {:x 0, :y 0, :w 5, :h 15}, :options {:viz/style-name "widget", :viz/y-title "power", :viz/x-title "frequency", :viz/allowDecimals false,
+                                                      :viz/banner-color {:r 0, :g 0, :b 255, :a 1}, :viz/tooltip {:followPointer true}, :viz/title "Channels (area)",
+                                                                                                                                                     :viz/banner-text-color {:r 255, :g 255, :b 255, :a 1}, :viz/animation false}}
+     {:ret_types [:data-format/x-y-n :data-format/x-y :data-format/x-y-e :data-format/y],
+      :key "2f029b5f-14a4-4b0f-986d-3a316795f603", :name :bubble-widget,
+      :username "chad1", :basis :chart, :data-source :bubble-service, :type :bubble-chart,
+      :icon "/images/bubble-widget.png", :label "Bubble", :data-grid {:x 0, :y 15, :w 5, :h 15},
+      :options {:viz/title "Bubble", :viz/banner-color {:r 0, :g 100, :b 0, :a 1}, :viz/banner-text-color {:r 255, :g 255, :b 255, :a 1}, :viz/dataLabels true, :viz/labelFormat "{point.name}", :viz/lineWidth 0, :viz/animation false, :viz/data-labels true}}))
+
+  (def data-ordered (map #(into (sorted-map) %) data))
+  data-ordered
+
+
+  (def data "({:ret_types [:data-format/x-y :data-format/x-y-n :data-format/x-y-e :data-format/y], :key \"07f9d07a-d315-4459-80f3-5e93ea0b4350\", :name :area-widget, :username \"chad2\", :basis :chart, :data-source :spectrum-traces, :type :area-chart, :icon \"/images/area-widget.png\", :label \"Area\", :data-grid {:x 0, :y 0, :w 5, :h 15}, :options {:viz/style-name \"widget\", :viz/y-title \"power\", :viz/x-title \"frequency\", :viz/allowDecimals false, :viz/banner-color {:r 0, :g 0, :b 255, :a 1}, :viz/tooltip {:followPointer true}, :viz/title \"Channels (area)\", :viz/banner-text-color {:r 255, :g 255, :b 255, :a 1}, :viz/animation false}} {:ret_types [:data-format/x-y-n :data-format/x-y :data-format/x-y-e :data-format/y], :key \"46e31202-a746-43b4-af53-549833e5188e\", :name :bubble-widget, :username \"chad2\", :basis :chart, :data-source :bubble-service, :type :bubble-chart, :icon \"/images/bubble-widget.png\", :label \"Bubble\", :data-grid {:x 1, :y 15, :w 5, :h 15}, :options {:viz/title \"Bubble\", :viz/banner-color {:r 0, :g 100, :b 0, :a 1}, :viz/banner-text-color {:r 255, :g 255, :b 255, :a 1}, :viz/dataLabels true, :viz/labelFormat \"{point.name}\", :viz/lineWidth 0, :viz/animation false, :viz/data-labels true}})")
```

Comment:
do we want ALL these comments for future debugging/reference?  I know I like to do this too, but i feel like we should maybe try to keep rich comments to just the last/end of file, not intermixed between legit functions.

Reply:
Yeah I agree, I kept all the files a mess until I got it working lol. I moved it to the bottom where rich comments belong.